recorrer.h

```diff
@@ -21,10 +21,15 @@ void recorrer(Iterator ItBegin, Iterator ItEnd, F ope)
 
 // #1
 template <typename Container, typename F>
-void recorrer(Container &container, F ope)
-{  
-    recorrer(container.begin(), container.end(), ope);
+void recorrer(Container &container, F ope,TX dir){  
```

Comment:
It's not necessary to add the dir argument.
Just send the begin and end iterator would be fine.